x-pack/legacy/plugins/apm/public/components/shared/Summary/TransactionSummary.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { idx } from '@kbn/elastic-idx';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { Summary } from './';
+import { TimestampSummaryItem } from './TimestampSummaryItem';
+import { DurationSummaryItem } from './DurationSummaryItem';
+import { ErrorCountSummaryItem } from './ErrorCountSummaryItem';
+import { isRumAgentName } from '../../../../common/agent_name';
+import { HttpInfoSummaryItem } from './HttpInfoSummaryItem';
+import { TransactionResultSummaryItem } from './TransactionResultSummaryItem';
+
+interface Props {
+  transaction: Transaction;
+  totalDuration: number | undefined;
+  errorCount: number;
+}
+
+const getTransactionResultSummaryItem = (transaction: Transaction) => {
+  const result = idx(transaction, _ => _.transaction.result);
+  const isRumAgent = isRumAgentName(transaction.agent.name);
+  const url = isRumAgent
+    ? idx(transaction, _ => _.transaction.page.url)
+    : idx(transaction, _ => _.url.full);
+
+  if (url) {
+    const method = idx(transaction, _ => _.http.request.method) || '';
```

Comment:
Perhaps make `method` optional, and then handle the undefined state in `HttpInfoSummaryItem`.

Reply:
My assumption was that `method` is always set when `url` is set, and I just need to trick TS. Is that assumption incorrect?

Reply:
I don't think transactions from the RUM agent has the `method` property. It creates a separate span for outgoing http requests.

Reply:
Good one, I'll check.